gumroad_utils/scrapper.py

```diff
@@ -227,7 +236,8 @@ def _traverse_tree(items: list[dict], tree_path: Path, parent_folder: Path) -> N
                 )
 
             for item in items:
-                if item["type"] != "file":
+                # NOTE(PxINKY) Gumroad added a "file" type that's embedded into the page resulting in no download_url
+                if item["type"] != "file" or item["download_url"] is None:
```

Comment:
Could you please add log, which file doesn't have a download URL,  here?

Reply:
Here are some examples i found:

**Google drive / External file host**
`{'type': 'file', 'file_name': 'Air Captain.spp - Google Drive', 'description': '', 'extension': 'URL', 'file_size': None, 'pagelength': None, 'duration': None, 'id': 'DhuqduTkneMdU1ksrgK5Kg==', 'download_url': None, 'stream_url': None, 'kindle_data': None, 'latest_media_location': None, 'content_length': None, 'read_url': None, 'external_link_url': 'https://drive.google.com/file/d/(Hiding extension due to Terms of Service)', 'subtitle_files': [], 'pdf_stamp_enabled': False, 'processing': False, 'thumbnail_url': None}`

**URL Embed**
`{'type': 'file', 'file_name': 'Avatars-3.0-Manager', 'description': '', 'extension': 'URL', 'file_size': None, 'pagelength': None, 'duration': None, 'id': 'SpLhtW9tbOdjB0eeq71Giw==', 'download_url': None, 'stream_url': None, 'kindle_data': None, 'latest_media_location': None, 'content_length': None, 'read_url': None, 'external_link_url': 'https://github.com/VRLabs/Avatars-3.0-Manager', 'subtitle_files': [], 'pdf_stamp_enabled': False, 'processing': False, 'thumbnail_url': None}`

**Youtube Embed**
`{'type': 'file', 'file_name': 'How to set up your hairstyle v2 ⭐ (Easy setup) - By AidenFur - YouTube', 'description': '', 'extension': 'URL', 'file_size': None, 'pagelength': None, 'duration': None, 'id': 'cM1_iVZaiUMamkMKZkaz_A==', 'download_url': None, 'stream_url': None, 'kindle_data': None, 'latest_media_location': None, 'content_length': None, 'read_url': None, 'external_link_url': 'https://www.youtube.com/watch?v=kAypVPhdUis&lc=Ugy8UnmwuaOeVQtU-Nt4AaABAg', 'subtitle_files': [], 'pdf_stamp_enabled': False, 'processing': False, 'thumbnail_url': None}`

That being said you defiantly could check/use `external_link_url`